supremm/outputter.py

```diff
@@ -21,6 +26,42 @@ def __exit__(self, exception_type, exception_val, trace):
         return self._impl.__exit__(exception_type, exception_val, trace)
 
 
+class FileOutput(object):
+    """
+    Simple outputter that dumps the job summary to a file given by out_file. Intended for debug purposes.
+    """
+    def __init__(self, outconf, resconf):
+        self._resid = resconf['resource_id']
+        self._path = outconf['out_file']
+        if not os.path.exists(self._path):
+            raise Exception("Path specified by out_file does not exist")
+        self._file = open(self._path, 'w')
+        self._dojson = True
+        if self._dojson:
+            self._jsonfile = open(self._path+'.json', 'w')
+            self._jsonarray = []
+
+    def __enter__(self):
+        print ("Calling enter on file", file=self._file)
+        return self
+
+    def process(self, summary, mdata):
+        """
+        json print
+        """ 
+        print(self._resid, json.dumps(summary.get(), indent=4), file=self._file)
+        print("MDATA: ", json.dumps(mdata, indent=4), file=self._file)
+        if self._dojson:
+            self._jsonarray.append(summary.get())
+            self._jsonarray.append(mdata)
+
+    def __exit__(self, exception_type, exception_val, trace):
+        print("Calling exit on {}".format(self._file.name), file=self._file)
+        self._file.close()
```

Comment:
For me printing to stdout like this hasen't called any crashes or behavior
I think is weird. I'm using it right and prints normally.

- Ian

On Mon, Aug 15, 2016 at 3:24 PM, Joe White <notifications@github.com> wrote:

> In supremm/outputter.py
> <https://github.com/ubccr/supremm/pull/42#discussion_r74818344>:
>
> > +        print ("Calling enter on file", file=self._file)
> > +        return self
> > +
> > +    def process(self, summary, mdata):
> > +        """
> > +        json print
> > +        """
> > +        print(self._resid, json.dumps(summary.get(), indent=4), file=self._file)
> > +        print("MDATA: ", json.dumps(mdata, indent=4), file=self._file)
> > +        if self._dojson:
> > +            self._jsonarray.append(summary.get())
> > +            self._jsonarray.append(mdata)
> > +
> > +    def __exit__(self, exception_type, exception_val, trace):
> > +        print("Calling exit on {}".format(self._file.name), file=self._file)
> > +        self._file.close()
>
> What happens if you call close() on sys.stdout? This was not being done in
> the previous StdoutOutputter implementation.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ubccr/supremm/pull/42/files/d1a6b26c7dc26dc33af1756c37c0cb38b80721cd#r74818344>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGyn2NEg-QaFivMWUpzM1FeHY1t0K4DXks5qgLzigaJpZM4JkoVB>
> .
>